src/main/java/com/example/entityapi/service/EntityLinkService.java

```diff
@@ -51,6 +52,11 @@ public EntityLink get(String id)
 	@Override
 	public EntityLink createUpdate(EntityLink entityLink)
 	{
+		if (entityLink.getId() == null || entityLink.getId().isBlank())
+		{
+			entityLink.setId(UUID.randomUUID().toString());
+		}
+
 		logger.debug("Create/update entity link: " + entityLink.getId());
```

Comment:
Ideally the log would say which of a create or update occurred.

Reply:
The controller layer will do this.